src/prefect/runner/storage.py

```diff
@@ -470,6 +472,68 @@ def __repr__(self) -> str:
         return f"RemoteStorage(url={self._url!r})"
 
 
+class BlockStorageAdapter:
+    """
+    A storage adapter for a storage block object to allow it to be used as a
+    runner storage object.
+    """
+
+    def __init__(
+        self,
+        block: ReadableDeploymentStorage,
+        pull_interval: Optional[int] = 60,
+    ):
+        self._block = block
+        self._pull_interval = pull_interval
+        self._storage_base_path = Path.cwd()
+        if not hasattr(block, "get_directory"):
+            raise ValueError(
+                "Provided block must implement the ReadableDeploymentStorage interface."
+            )
+
+        self._name = (
+            f"{block.get_block_type_slug()}-{block._block_document_name}"
+            if block._block_document_name
+            else str(uuid4())
+        )
+
+    def set_base_path(self, path: Path):
+        self._storage_base_path = path
+
+    @property
+    def pull_interval(self) -> Optional[int]:
+        return self._pull_interval
+
+    @property
+    def destination(self) -> Path:
+        return self._storage_base_path / self._name
+
+    async def pull_code(self):
+        await self._block.get_directory(local_path=str(self.destination))
+
+    def to_pull_step(self) -> dict:
+        # Give blocks the change to implement their own pull step
+        if hasattr(self._block, "get_pull_step"):
+            return self._block.get_pull_step()
+        else:
+            if not self._block._block_document_name:
+                raise BlockNotSavedError(
```

Comment:
Yes! I'm glad you implemented it this way, I want us to move away from auto-saving blocks as I think it creates clutter.

Reply:
+1 as I'm about to clean up 775k `Local File System` blocks for a customer this morning ;)